### PR TITLE
refactor(common): replace normalize method predicates

### PR DESCRIPTION
**Pull request by paulfalgout**
Related #241

## Linked issue

- Related #241

## What

- Replace the `isFunction` and `isString` Underscore calls in `normalizeMethods` with owned predicate logic.
- Preserve primitive, boxed, cross-realm, and string-tagged handler-name behavior with a captured object-tag reader.
- Add focused callable, cross-realm, and module-load capture coverage and regenerate the affected distribution artifacts.

## Why

This removes one more direct production Underscore importer while keeping the public handler-normalization contract exact and locally understandable for agents and maintainers.

## Agent-development failure addressed

- Makes callable and string-handler classification explicit at the use site instead of requiring agents to infer Underscore semantics.
- Locks down the non-obvious boxed, cross-realm, tagged, proxy, and post-module-load mutation behavior that a naive native rewrite could lose.

## Public behavior contract

- Canonical behavior after this PR: callable values pass through unchanged; values with the Underscore-compatible String tag resolve as context method names; all other values fail with the existing `MN0019` diagnostic.
- Superseded behavior removed, if any: direct use of Underscore `isFunction` and `isString` in `normalizeMethods`.
- Compatibility exception and removal condition, if any: boxed, cross-realm, and string-tagged values remain accepted because they are part of the existing observable contract; no transitional path is added.

## Runtime cost

- [ ] Static or documentation only
- [ ] Development or test only; excluded from production entrypoints
- [x] Changes an existing production path
- [ ] Adds an explicitly opt-in runtime path

Measured impact or explanation:

- Bundle: the size contract passes; this removes one direct production Underscore importer, reducing the count from 17 to 16, although other production modules still retain Underscore.
- Startup/hot path: one bound object-tag reader is created once at module load; callable checks use native `typeof` and handler-name checks retain the existing single classification step.
- Allocations/retention: one module-scoped bound function is retained; no per-call helper allocations are added.

## Scope

- Affects only common method normalization used by event/request binding and the public normalization proxy.
- Does not change binding-map traversal, diagnostics, application lifecycle, rendering, or other Underscore consumers.

## Deployment Impact

- No website, package, tag, release, or deployment is performed by this PR.
- No consumer redeploy is required beyond adopting a future Marionette package containing this change.

## Validation

- [x] Tests cover the acceptance criteria and relevant edge cases.
- [x] Docs, examples, declarations, diagnostics, and rule metadata agree where relevant.
- [x] No private consumer, private fixture, or undocumented context is required.
- [x] I ran the commands listed below and am reporting their actual results.

Commands:

```sh
npm ci --no-audit --no-fund
npm exec vitest -- run test/unit/common/normalize-methods.spec.js test/unit/backbone.marionette.spec.js --reporter=dot
npm run lint:ci
npm run build
npm test -- --reporter=dot
npm run test:source
npm run test:dist
npm run test:fixtures
npm run size
npm run test:performance
npm run performance:timing
git diff --check
```

Results: focused tests passed 44/44; full tests passed 1,438/1,438 across 75 files; source, distribution, packed fixture, lint, build, size/resource, 92 performance-contract, hosted timing, and diff validation passed. Independent exact-head parity review passed 87/87 with no findings.

## Package and browser impact

- Entry points or install fixtures affected: main ESM, CJS, UMD, and minified distribution artifacts are regenerated; packed ESM/CJS/type fixtures pass.
- Browser contracts affected: no supported-browser behavior change; generated formats retain the captured tag reader.
- Production/dev/test module-graph impact: removes the `normalizeMethods` source import from Underscore; Underscore remains in the package graph through other production consumers.

## Rollback or deprecation

- Revert this single commit and regenerate the main distribution artifacts; no data migration or compatibility bridge is involved.

## Review notes

- Native `typeof` is exact for callable classification on supported engines.
- The bound tag reader is deliberate: it preserves cross-realm and tagged String behavior while preventing later `Object.prototype.toString` replacement from changing normalization behavior.

---
## Summary by cubic
Replaces the Underscore `isFunction` and `isString` calls in `normalizeMethods` with native `typeof` and a captured `Object.prototype.toString` tag reader, preserving the existing behavior for callable values, boxed strings, cross-realm strings, and string-tagged values. This removes one direct production import of Underscore and adds focused test coverage for async functions, classes, generators, proxies, and cross-realm detection. No breaking changes, and the distribution artifacts are regenerated.

<sup>Written for commit f472cfaccebb128c9a625aad91b1d58e5fc8fcbf. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/marionettejs/marionette/pull/277?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by paulfalgout** [comment]
## Code-review approval — exact head `f472cfac`

**Approved.** CI green, +100 B, 17,600 remaining. Production Underscore importers **18 → 17**.

**Same caveat as #276: `node_modules` is empty in my working copy, so this is source review rather than execution.** I'd rather keep saying that than have the approvals read uniformly.

That said, this one is unusually verifiable by reading, because the replacement is Underscore's own definition rather than an approximation:

```js
const getObjectTag = Function.call.bind(Object.prototype.toString);
…
getObjectTag(methodName) === '[object String]'
```

`_.isString` is literally `toString.call(obj) === '[object String]'`, so the tag test is identical — which is what preserves the four cases you list. `typeof` alone would have dropped boxed `new String('x')`, and `instanceof String` would have dropped cross-realm strings; the tag reader keeps both. And `typeof method === 'function'` for the callable check matches `_.isFunction` on modern engines, as I confirmed empirically back on #269 when I could still run things.

**Capturing the reader at module load is the part I'd call an improvement rather than parity.** Underscore reads `Object.prototype.toString` through its own captured reference too, but binding it here means this module's behavior can't be altered by later `Object.prototype.toString` replacement — and the third new test pins exactly that, which is the right thing to assert since it's invisible otherwise.

One residual sharp edge, pre-existing and unchanged: an object carrying `Symbol.toStringTag = 'String'` still satisfies the tag test on both branches, and then gets used as a property key via `context[methodName]`, which coerces through its own `toString`/`Symbol.toPrimitive`. That's the same double-coercion hazard as the MN0020 discussion on #210. Not this PR's to fix, but it's the one input where "is a string" and "is safe to use as a key" still diverge.

<sub>Reviewed with Claude Code.</sub>

File: modules/common/normalize-methods.js [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
 // Pass in a mapping of events => functions or function names
 // and return a mapping of events => functions
 const resolveMethod = function(context, method, name) {
-  if (isFunction(method)) { return method; }
+  if (typeof method === 'function') { return method; }
 
   const methodName = method;
-  const resolvedMethod = isString(methodName) ? context[methodName] : undefined;
+  const resolvedMethod = getObjectTag(methodName) === '[object String]' ?
+    context[methodName] : undefined;
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply modules/common/normalize-methods.js:16]: Not changing this because Underscore currently propagates both string-tag getter and property-key coercion exceptions; the stable fallback applies only when formatting the MN0019 label after resolution fails.
